api/src/main/java/org/openmrs/api/impl/ObsServiceImpl.java

```diff
@@ -213,9 +216,35 @@ private Obs saveObsNotDirty(Obs obs, String changeMessage) {
 	private Obs saveNewOrVoidedObs(Obs obs, String changeMessage) {
 		Obs ret = dao.saveObs(obs);
 		saveObsGroup(ret,changeMessage);
+		saveObsReferenceRange(obs);
 		return ret;
 	}
 
+	private void saveObsReferenceRange(Obs obs) {
+		Concept concept = obs.getConcept();
+		ConceptService conceptService = Context.getConceptService();
+		ObsService obsService = Context.getObsService();
+
+		if (concept != null) {
+
+			List<ConceptReferenceRange> conceptReferenceRanges = conceptService.getConceptReferenceRangesByConceptId(concept.getId());
```

Comment:
Should you do all this even when the concept data type is not Numeric?

Reply:
Good point, let me add a check.